Voxel r2s

zxkjack123 opened the pull request:
Main changes:

- Remove not used mat_* tag of class mesh in mesh.py

- Add a parameter `sub_voxel` that enable user to choose sub-voxel r2s

- Duplicate neutron flux for each sub-voxel in ALARA fluxin

- Eliminate  fluxes of void sub-voxel from ALARA fluxin

- Add a test function to test ALARA fluxin in pyne/tests/test_alara.py

- Add two files of expected ALARA fluxin in pyne/tests/files_test_alara/

zxkjack123 committed:
20171026 remove not used mat_*tag edit 7

zxkjack123 committed:
Merge branch 'develop' of https://github.com/pyne/pyne into voxel_r2s

zxkjack123 committed:
add parameter sub_voxel

zxkjack123 committed:
sub-voxel r2s: duplicate flux for sub-voxel

zxkjack123 committed:
eliminate void sub-voxel from ALARA flux

scopatz commented on pyne/alara.py:
```diff
@@ -48,6 +49,30 @@ def mesh_to_fluxin(flux_mesh, flux_tag, fluxin="fluxin.out",
+    sub_voxel: bool, optional
+        If true, sub-voxel r2s work flow will be sued. Flux of a voxel will
+        be duplicated c times. Where c is the cell numbers of that voxel.
+    cell_fracs : structured array
```
This should be listed as optional

scopatz commented on pyne/alara.py:
```diff
@@ -48,6 +49,30 @@ def mesh_to_fluxin(flux_mesh, flux_tag, fluxin="fluxin.out",
+
+        The array must be sorted with respect to both idx and cell, with
+        cell changing fastest.
+    cell_mats : dict
```
This should also be listed as optional

scopatz commented on pyne/alara.py:
```diff
@@ -70,18 +95,35 @@ def mesh_to_fluxin(flux_mesh, flux_tag, fluxin="fluxin.out",
+
+            output += "\n\n"
+
+    if sub_voxel:
```
The should be an `else:`

scopatz commented:
LGTM after a couple of modifications!

zxkjack123 committed:
Update alara.py

1. cell_fracs and cell_mats are marked as "optional"
2. if sub_voxel is replaced with else.

zxkjack123 commented:
There are 3 changes according to the comments of @scopatz .

scopatz commented on pyne/alara.py:
```diff
@@ -70,18 +95,35 @@ def mesh_to_fluxin(flux_mesh, flux_tag, fluxin="fluxin.out",
+                        if count % 6 == 0:
+                            output += "\n"
+
+                    output += "\n\n"
```
Since the body of these two clause is mostly the same, can you maybe please provide a private method (ie a method that begins with an underscore `_`) or similar that performs the work.  This function would then just be called with different arguments.  There is a lot of code duplication here that could be removed :) Thanks!

zxkjack123 replied:
Thanks for your suggestion @scopatz . I put these two clause into a private method `_output_flux()` to reduce duplication.

zxkjack123 committed:
reduce the duplication of mesh_to_fluxin

zxkjack123 committed:
Update alara.py

Add a new line at end of file

scopatz commented on pyne/alara.py:
```diff
@@ -704,3 +727,24 @@ def cram(N, t, n_0, order):
+    :param tag_flux:
+    :param output: string
+    :return: string
+    """
```
Please write this docstring in numpydoc format, so that it conforms to the other docstrings throughout pyne. Thanks!

zxkjack123 replied:
I rewrote this docstring in numpydoc format.

zxkjack123 committed:
update docstring of _output_flux

scopatz commented:
Thanks @zxkjack123!
